cpp/src/io/orc/stripe_data.cu

```diff
@@ -1340,21 +1331,17 @@ static __device__ void DecodeRowPositions(orcdec_state_s *s, size_t first_row, i
       // TBD: Brute-forcing this, there might be a more efficient way to find the thread with the
       // last row
       last_row = (nz_count == s->u.rowdec.nz_count) ? row_plus1 : 0;
-      last_row = max(last_row, SHFL_XOR(last_row, 1));
-      last_row = max(last_row, SHFL_XOR(last_row, 2));
-      last_row = max(last_row, SHFL_XOR(last_row, 4));
-      last_row = max(last_row, SHFL_XOR(last_row, 8));
-      last_row = max(last_row, SHFL_XOR(last_row, 16));
+      for (int width = 1; width <= 16; width *= 2) {
+        last_row = max(last_row, shuffle_xor(last_row, width));
+      }
```

Comment:
You should be able to use [this](https://nvlabs.github.io/cub/classcub_1_1_warp_reduce.html#a0dd72fc4cf7e1ecf59e8b15bd6819185) here

Reply:
Yep. Had missed that one.